daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilParseOutputStreamContentHandler.scala

```diff
@@ -107,6 +208,36 @@ class DaffodilParseOutputStreamContentHandler(out: OutputStream, pretty: Boolean
     outputNewlineStack.push(false)
   }
 
+  private def outputTagName(uri: String, localName: String, qName: String, atts: Maybe[Attributes] = Nope): Unit = {
+    val tagName = {
+      if (qName.nonEmpty) {
+        qName
+      } else {
+        // use attributes
+        lazy val attrs = atts.get
+        lazy val in = attrs.getIndex(uri, localName)
+        if (atts.nonEmpty && in >= 0) {
+          val qn = attrs.getQName(in)
+          qn
+        } else {
+          // use globalPrefixMapping
+          val sanitizedUri = if (uri.isEmpty) null else uri
+          val optPrefix = XMLUtils.getPrefixOption(globalPrefixMapping, sanitizedUri)
+          optPrefix match {
+            case Some(pre) =>
+              if (pre != null && pre.nonEmpty) {
+                s"$pre:$localName"
+              } else {
+                localName
+              }
+            case None => localName
```

Comment:
Does this case imply something went wrong? I guess this would mean we never got a prefix mapping for some URI, but then saw an element with that URI? I'm not sure what SAX says about how to handle this. I could see an argument that this is a SAXException or something rather than silenting ignoring the uri, but maybe that's being too strict?